module/VuFind/src/VuFind/Connection/LibGuides.php

```diff
@@ -0,0 +1,192 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Connection
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Brent Palmer <brent-palmer@icpl.org>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+
+namespace VuFind\Connection;
+
+use Exception;
+use Laminas\Log\LoggerAwareInterface;
+
+/**
+ * LibGuides API connection class.
+ *
+ * Note: This is for the LibGuides API used by the LibGuidesProfile recommendation service,
+ * this is *not* for the LibGuides search widget "API" used by the LibGuides and LibGuidesAZ
+ * data sources.
+ *
+ * Closely adapted from VuFind\DigitalContent\OverdriveConnector.
+ *
+ * @category VuFind
+ * @package  Connection
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Brent Palmer <brent-palmer@icpl.org>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+class LibGuides implements
+    OauthServiceInterface,
+    \VuFindHttp\HttpServiceAwareInterface,
+    LoggerAwareInterface
+{
+    use OauthServiceTrait;
+    use \VuFindHttp\HttpServiceAwareTrait;
+    use \VuFind\Log\LoggerAwareTrait {
+        logError as error;
+    }
+
+    /**
+     * HTTP Client
+     *
+     * @var \Laminas\Http\HttpClient
+     */
+    protected $client;
+
+    /**
+     * Base URL of the LibGuides API
+     *
+     * @var string
+     */
+    protected $baseUrl;
+
+    /**
+     * Client ID for a client_credentials grant
+     *
+     * @var string
+     */
+    protected $clientId;
+
+    /**
+     * Client Secret for a client_credentials grant
+     *
+     * @var string
+     */
+    protected $clientSecret;
+
+    /**
+     * Constructor
+     *
+     * @param \Laminas\Http\Client $client             HTTP client
+     * @param string               $baseUrl            Base URL of the LibGuides API
+     * @param string               $clientId           Client Id provided by LibGuides API configuration
+     * @param string               $clientSecret       Client Secret provided by LibGuides API configuration
+     * @param bool                 $forceNewConnection Force a new connection (get a new token)
+     *
+     * @link https://ask.springshare.com/libguides/faq/873#api-auth
+     */
+    public function __construct(
+        $client,
+        $baseUrl,
+        $clientId,
+        $clientSecret,
+        $forceNewConnection = false
```

Comment:
Yeah phpstan, I know I'm not using this yet.  I have to see in what cases the OverdriveConnector actually forced a new connection.

Reply:
So it turns out OverdriveConnector never passed this boolean as true either!  That will teach me to copy/past :)  Removed altogether.